SkillsFunctionalTests/tests/SkillFunctionalTests/Configuration/BotTestConfiguration.cs

```diff
@@ -2,33 +2,55 @@
 // Licensed under the MIT License.
 
 using System;
+using Microsoft.Extensions.Configuration;
+using Xunit.Sdk;
 
 namespace SkillFunctionalTests.Configuration
 {
-    public class BotTestConfiguration : IBotTestConfiguration
+    /// <summary>
+    /// Load test configuration from appsettings.json, appsettings.Development.json (if present) and environment variables.
+    /// </summary>
+    /// <remarks>
+    /// This class will initialize the test configuration from config or environment variables.
+    /// The values in appsettings will be overriden by the ones in appsettings.Development (if present) and those
+    /// will be also overriden by the values in environment variables (if set).
+    /// </remarks>
+    public class BotTestConfiguration
     {
-        public BotTestConfiguration(string directLineSecret, string botId)
+        private const string DirectLineSecretKey = "DIRECTLINE";
+        private const string BotIdKey = "BOTID";
+        private static readonly IConfiguration _configuration;
+
+        /// <summary>
+        /// Static constructor to initialize IConfiguration only once.
+        /// </summary>
+        static BotTestConfiguration()
+        {
+            _configuration = new ConfigurationBuilder()
+                .AddJsonFile("appsettings.json")
+                .AddJsonFile("appsettings.Development.json", true, true)
+                .AddEnvironmentVariables()
+                .Build();
+        }
+
+        public BotTestConfiguration()
         {
-            if (string.IsNullOrEmpty(botId))
+            // DirectLine secret
+            DirectLineSecret = _configuration[DirectLineSecretKey];
+            if (string.IsNullOrWhiteSpace(DirectLineSecret))
             {
-                throw new ArgumentNullException(nameof(botId));
+                throw new XunitException($"Configuration setting '{DirectLineSecret}' not found.");
```

Comment:
Is there any reason why this is a _XunitException_ but the one below is an _ArgumentException_?

Reply:
Seems like we have both in different parts of the code. We should unify it. I will start a PR for that purpose.

Reply:
That was an error on my part, let me fix in a different PR, thanks for bringing this up.